### PR TITLE
Add __repr__ for all simple polynomials, add const property to Constant.

### PR DESCRIPTION
**Pull request by philippeitis**
I haven't implemented __repr__ for the classes instantiated by monomials, since implementing that there would require storing the monomials.

### CODE REVIEW

File: tests/test_polynomials_repr.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+"""Unit-testing module defining polynomials __str__ test cases."""
```
Comment (allexks) [inline tests/test_polynomials_repr.py]: I believe you just forgot to change `__str__` to `__repr__` here

File: tests/test_polynomials_repr.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    LinearBinomial,
+    Monomial,
+    Polynomial,
+    QuadraticTrinomial,
+    ZeroPolynomial,
+)
+
+
+class TestPolynomialsRepr(unittest.TestCase):
+    """Defines polynomials __str__ test cases."""
```
Comment (allexks) [inline tests/test_polynomials_repr.py]: Here as well

File: tests/test_polynomials_repr.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        self.assertEqual(expect, r)
+
+    def test_internal_zero_kept(self):
+        """Test that constants with value 1 are displayed properly."""
+        coeffs = [1, 0, 1]
+        expect = "Polynomial(1, 0, 1)"
+
+        r = repr(Polynomial(coeffs))
+
+        self.assertEqual(expect, r)
```
Comment (allexks) [inline tests/test_polynomials_repr.py]: I think that these few test cases now are redundant. Testing `__str__` has all these specific things that might come up as bugs but there is nothing like that with `__repr__` (talking about `test_negative_coefficient_unchanged`,  `test_trailing_zero_kept`, `test_constant_kept` and `test_internal_zero_kept`). I can imagine at most testing `repr(Polynomial(0, 0, 0))` or something like that as an edge case but I think the mentioned tests above are totally unnecessary and irrelevant as is the case now with `__repr__`'s implementation.
